spec/input_delegator_spec.rb

```diff
@@ -0,0 +1,12 @@
+require_relative '../lib/parser.rb'
+
+RSpec.describe(Parser) do
+    context "When creating a Parser" do
+        it "should instantiate a Parser object", positive: true do
+            parser = Parser.new
+            expect(parser).to_not be(nil)
+        end
+    end
+
+    # I am not sure how to test this.
```

Comment:
I notice that the `Parser#parse` method from above is pretty complicated. That might be part of the reason it is hard to test. I would be happy to pair with you on this if you have some time after Thanksgiving (and you are still working on this project)